pkg/controller/windowsmachineconfig/nodeconfig/nodeconfig.go

```diff
@@ -48,6 +51,56 @@ func (nc *nodeConfig) Configure() error {
 	if err := nc.handleCSRs(); err != nil {
 		return errors.Wrap(err, "handling CSR for the given node failed")
 	}
+
+	var instanceID string
+	// As the CSR has been approved get the Kubernetes node object associated
+	// the Windows VM created
+	if nc.Windows.GetCredentials() != nil && nc.Windows.GetCredentials().GetInstanceId() != "" {
+		instanceID = nc.Windows.GetCredentials().GetInstanceId()
+		if instanceID == "" {
+			return errors.Errorf("unable to get instanceID ")
+		}
+	}
+
+	// Get the node object associated with node
+	node, err := nc.getNodeObjectAssociated(instanceID)
+	if err != nil {
+		return errors.Wrap(err, "cannot get node object associated with the node")
+	}
+
+	// Apply worker labels
+	if err := nc.applyWorkerLabel(node); err != nil {
+		return errors.Wrap(err, "failed applying worker label")
+	}
+	return nil
+}
+
+func (nc *nodeConfig) getNodeObjectAssociated(instanceID string) (*v1.Node, error) {
+	nodes, err := nc.k8sclientset.CoreV1().Nodes().List(metav1.ListOptions{LabelSelector: "node.openshift.io/os_id=Windows"})
+	if err != nil {
+		return nil, errors.Wrap(err, "error listing nodes")
+	}
+	for _, node := range nodes.Items {
+		if instanceID == getInstanceID(node.Spec.ProviderID) {
+			return &node, nil
+		}
+	}
+	return nil, errors.Errorf("unable to find node for instanceID %s", instanceID)
+}
+
+// getInstanceID gets the instanceID of VM for a given cloud provider ID
+// Ex: aws:///us-east-1e/i-078285fdadccb2eaa. We always want the last entry which is the instanceID
+func getInstanceID(providerID string) string {
```

Comment:
nit: I want to explain a pattern that I've noticed wrt testing, if you are writing functionality for A and B. TestA and TestB represent testing functions associated with A and B.

A -> TestA should not share any code with A, meaning don't call the very method you are testing to validate.
B -> TestB should not share any code with B, same as above

TestA -> can have methods that are part of B. This will be case where you already have functionality for B and you have to duplicate a lot of code in TestA unless you use it from B, so this ok if the method is already public. In this case, it would have been ok say if `GetInstanceID` is already public because it is needed by another package.

Reply:
Kept it as private as we are currently not using it outside package. We can always change it to public when needed.